gremlin/audio_player.py

```diff
@@ -14,31 +14,66 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from typing import List
 
 from PySide6 import QtCore
 from PySide6.QtCore import QUrl
+from PySide6.QtMultimedia import QAudioOutput, QMediaPlayer
 
 from gremlin.common import SingletonDecorator
-from PySide6.QtMultimedia import QAudioOutput, QMediaPlayer
+from gremlin.config import Configuration
+
+
+class PlayListItem:
+    def __init__(self, sound_file: str, play_volume: int) -> None:
+        self.sound_file = sound_file
+        self.play_volume = play_volume
 
 
 @SingletonDecorator
 class AudioPlayer(QtCore.QObject):
-
     """Manages the playing of audio files."""
 
     def __init__(self) -> None:
         QtCore.QObject.__init__(self)
         self.player_output = QAudioOutput()
+
         self.player = QMediaPlayer()
+        self.player.mediaStatusChanged.connect(self._media_status_changed)
+
         self.player.setAudioOutput(self.player_output)
+        self.play_list: List[PlayListItem] = []
 
     def play(self, sound_filename: str, volume: int) -> None:
         """ volume: 0 = mute, >0 = volume level, max volume is 100 """
 
-        if self.player.playbackState() != QMediaPlayer.StoppedState:
+        sequential_play = Configuration().value("action", "play-sound", "sequential-play")
+        if (
+            sequential_play and self.player.playbackState() != QMediaPlayer.StoppedState
+        ):
+            playlist_item: PlayListItem = PlayListItem(sound_filename, volume)
+            self.play_list.append(playlist_item)
```

Comment:
Probably find to just in place construct and append the item.

Reply:
n/p